src/components/autocomplete-grid/autocomplete-grid/autocomplete-grid.ts

```diff
@@ -0,0 +1,285 @@
+      Array.from(this.querySelectorAll?.('sbb-autocomplete-grid-row')).filter(
+        (row) => !row.hasAttribute('data-disabled'),
+      ) ?? []
+    );
+  }
+
+  /** When an option is selected, update the input value and close the autocomplete. */
+  protected onOptionSelected(event: CustomEvent): void {
+    const target = event.target as SbbAutocompleteGridOptionElement;
+    if (!target.selected) {
+      return;
+    }
+
+    // Deselect the previous options
+    this.options
+      .filter((option) => option.id !== target.id && option.selected)
+      .forEach((option) => (option.selected = false));
+
+    if (this.triggerElement) {
+      // Set the option value
+      this.triggerElement.value = target.value as string;
+
+      // Manually trigger the change events
+      this.triggerElement.dispatchEvent(new Event('change', { bubbles: true }));
+      this.triggerElement.dispatchEvent(new InputEvent('input', { bubbles: true, composed: true }));
+    }
+
+    this.close();
+  }
+
+  protected onOptionClick(event: MouseEvent): void {
+    if (
+      (event.target as Element).tagName !== 'SBB-AUTOCOMPLETE-GRID-OPTION' ||
+      (event.target as SbbOptionElement).disabled
+    ) {
+      return;
+    }
+    this.close();
+  }
+
+  public override connectedCallback(): void {
+    super.connectedCallback();
+    if (ariaRoleOnHost) {
+      this.id ||= this.overlayId;
+    }
+    const signal = this.abort.signal;
+    this.addEventListener(
+      'autocompleteOptionSelectionChange',
+      (e: CustomEvent<void>) => this.onOptionSelected(e),
+      { signal },
+    );
+  }
+
+  protected syncNegative(): void {
+    this.querySelectorAll?.<SbbDividerElement | SbbAutocompleteGridButtonElement>(
+      'sbb-divider, sbb-autocomplete-grid-button',
+    ).forEach((e) => (e.negative = this.negative));
+
+    this.querySelectorAll?.<SbbAutocompleteGridOptionElement | SbbOptGroupElement>(
+      'sbb-autocomplete-grid-row, sbb-autocomplete-grid-option, sbb-autocomplete-grid-optgroup',
+    ).forEach((element) => element.toggleAttribute('data-negative', this.negative));
+  }
+
+  protected openedPanelKeyboardInteraction(event: KeyboardEvent): void {
+    if (this.state !== 'opened') {
+      return;
+    }
+
+    switch (event.key) {
+      case 'Escape':
+      case 'Tab':
+        this.close();
+        break;
+
+      case 'Enter':
+        this.selectByKeyboard(event);
+        break;
+
+      case 'ArrowDown':
+      case 'ArrowUp':
+        this.setNextActiveOption(event);
+        break;
+
+      case 'ArrowRight':
+      case 'ArrowLeft':
+        this._setNextHorizontalActiveElement(event);
+        break;
+    }
+  }
+
+  protected selectByKeyboard(event: KeyboardEvent): void {
+    if (this._activeColumnIndex !== 0) {
+      (
+        this._row[this._activeItemIndex].querySelectorAll(
+          'sbb-autocomplete-grid-option, sbb-autocomplete-grid-button',
+        )[this._activeColumnIndex] as SbbAutocompleteGridButtonElement
```

Comment:
I think I still don't understand :). what happens if a grid-option have been found and you apply dispatchClick() on it. Does it break?

Reply:
No prob :)
it should never happen to have a grid-option due to `_activeColumnIndex` value; but if it happens, yes, since dispatchClick is a method only available for the grid-button.


- horizontal navigation: `_activeColumnIndex` starts from 0 (option), then I group both option/buttons. If i stop on a button, `_activeColumnIndex` is 1, 2, 3.. otherwise back to 0 for option (see image)
- selectByKeyboard: I check for `_activeColumnIndex !== 0` before casting to button, so i'm sure it's not an option

![image](https://github.com/lyne-design-system/lyne-components/assets/101575400/15e934fc-69d3-4ef0-9213-b5c0eee2abc5)

Reply:
Sorry it took me a little long this time, but I understand now :) It makes sense. But could you maybe add a short explaining comment in the code as it is a little bit special?

Reply:
Comment added 👍